lib/src/parser.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:test_cov_console/src/parser_constants.dart';
+
+class Parser {
+  final List<String> arguments;
+
+  Parser(this.arguments);
+
+  Map<String, dynamic> parse() {
+    if (arguments.isEmpty) {
+      return {};
+    }
+
+    final Map<String, dynamic> result = {};
+    bool bContinue = false;
+    for (var i = 0; i < arguments.length; ++i) {
+      if (bContinue) {
+        bContinue = false;
+        continue;
+      }
+      final arg = arguments[i].toLowerCase();
+      if (arg == '-h' || arg.startsWith('--${ParserConstants.help}')) {
+        return _returnHelp();
+      }
+      try {
+        for (final opt in ParserConstants.fieldMap) {
+          if (arg == '-${opt.opt1}' || arg.startsWith('--${opt.opt2}')) {
+            if (arg == '-${opt.opt1}') {
```

Comment:
extract method with proper naming to avoid nested conditionals and to make code more readable, at current stage I'm not sure what the code is doing

Reply:
Updated please checked.